Doc: AVD beginner TOI update to be consistent with AVD examples (#1908)
aristanetworks/ansible-avd: ansible_collections/arista/avd/docs/getting-started/intro-to-ansible-and-avd.md

## What are the requirements to run Ansible?

Ansible can run on almost anything, but in production scenarios Ansible is typically deployed on a virtual Linux server,

> running on the customer’s preferred hypervisor. This Ansible server then communicates either directly with the

Reviewer: This will fail the ansible sanity test.

Author: Updated

Arista network devices via eAPI or with Arista CloudVision Portal, which in turn communicates with the Arista network devices.

> Controlling what Ansible does is typically done using an SSH terminal session to the Ansible server from the Operator’s computer.

Reviewer: This will fail the ansible sanity test.

Author: Updated

## When and when not to use AVD

> It’s important to note when and perhaps more importantly when not to use AVD.

Reviewer: This will fail the ansible sanity test.

Author: Updated

This is defined in the Ansible ***inventory***. Per the official Ansible documentation the [inventory](https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html#inventory-basics-formats-hosts-and-groups)
is a file typically in either INI or YAML format, but other inventory sources can be used. In Ansible AVD we use YAML format by default,
hence the file is called inventory.yml, located in the Ansible AVD project folder (in this case `single-dc-l3ls`):

![Figure: Ansible Inventory Folder Structure](../_media/getting-started/inventory-folder-structure.png)

> Please note that the example above is taken from the Ansible AVD Examples repository, specifically the `Single-DC-L3LS`.

Reviewer 2: Do we need to link to this?

Author: Updated

As stated earlier this file is in [YAML](https://yaml.org) format.

It's important that the hostnames specified in the inventory exist either in DNS or in the hosts file on your Ansible host to allow successful name lookup and be able to reach the switches directly. To test this, you must be able to successfully ping the host, for example, `ping dc1-spine1` from your Ansible host. Alternatively, if there is no DNS available, or if devices need to be reached using a fully-qualified domain-name (FQDN) define `ansible_host` to be an IP address or FQDN for each device - see dc1-spine1/2 below for an example:

```yml
---
all:
  children:
    FABRIC:
      children:
        DC1:
          children:
            DC1_SPINES:
              hosts:
                dc1-spine1:
                  ansible_host: 172.16.1.11
                dc1-spine2:
                  ansible_host: 172.16.1.12
```

> However, going forward we expect that all hostnames specified are reachable without this workaround, hence the entire inventory file looks as follows:

Reviewer 3: I would not call `ansible_host` a workaround, as described in the doc, there are legit usage for this, if for instance you want to use aliases in your inventory file

Author: Changed

```yml
---
all:
  children:
    FABRIC:
      children:
        DC1:
          children:
            DC1_SPINES:
              hosts:
                dc1-spine1:
                dc1-spine2:
            DC1_L3_LEAVES:
              hosts:
                dc1-leaf1a:
                dc1-leaf1b:
                dc1-leaf2a:
                dc1-leaf2b:
            DC1_L2_LEAVES:
              hosts:
                dc1-leaf1c:
                dc1-leaf2c:

        NETWORK_SERVICES:
          children:
            DC1_L3_LEAVES:
            DC1_L2_LEAVES:
        CONNECTED_ENDPOINTS:
          children:
            DC1_L3_LEAVES:
            DC1_L2_LEAVES:
```

Don’t confuse ***hosts*** with servers or similar. A host can be anything that can be accessed via SSH or an API, to be managed by Ansible,
including Arista switches.

The settings inside the inventory.yml file are defined in a tree-like structure using ***groups***. Groups can contain hosts or other groups -
making it easier to apply common configuration to a group of devices.

The ***all*** line at the top is a default group that contains all ***hosts*** i.e. all switches. Don’t worry too much about that for now.

> The ***children:*** keyword is used to define “groups of groups” i.e. just an internal keyword to differentiate between hosts and groups.

Reviewer 3: Same comment as Tony regarding the change of quotes and double-quotes, better keep the ASCII charcters in there

Author: Should be updated, please take another look

ansible_user: ansible
ansible_password: ansible
# User escalation (to enter enable mode)
ansible_become: true
ansible_become_method: enable
# Use SSL (HTTPS)
ansible_httpapi_use_ssl: true
# Do not try to validate certs
ansible_httpapi_validate_certs: false

# Common AVD group variables
fabric_name: FABRIC

# Local users
local_users:
  # Define a new user, which is called "ansible"
  ansible:
    privilege: 15
    role: network-admin
    # Password set to "ansible". Same string as the device generates when configuring a username.
    sha512_password: $6$7u4j1rkb3VELgcZE$EJt2Qff8kd/TapRoci0XaIZsL4tFzgq1YZBLD9c6f/knXzvcYY0NcMKndZeCv0T268knGKhOEwZAxqKjlMm920
  admin:
    privilege: 15
    role: network-admin
    no_password: true
```

The above snippet defines a number of defaults that apply to all children of FABRIC along with configuration specific to
each leaf pair and finally each individual leaf switch.

Group variables can be overridden by specifying host variables for specific devices.
(See [DEFAULT_HASH_BEHAVIOUR](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#default-hash-behaviour)).
The order of precedence is (from lowest to highest):

> - 'All' group (because it is the ‘parent’ of all other groups).

Reviewer 3: ```suggestion
- 'All' group (because it is the 'parent' of all other groups).
```

Author: Updated

- Parent group.
- Child group.

> - Host.

Reviewer 3: Not sure dots are needed in this case

Author: Updated

another on your CloudVision instance, and another on each EOS device you look to configure.

At a minimum, each play defines two things:

> 1. The managed devices (***hosts***) to target, referenced from the ***inventory*** we defined earlier.

Reviewer 3: ```suggestion
1. The managed device(s) (***hosts***) to target, referenced from the ***inventory*** we defined earlier.
```

Author: Updated

> 2. One or more ***tasks*** to execute on the targets defined.

Reviewer 3: ```suggestion
2. One or more ***tasks*** to execute on the target(s) defined.
```

Author: Updated

## How do I use AVD?

AVD comes with pre-built templates that you can either use as-is or adapt to your liking.

Once the templates reflect your desired network configuration, you deploy the configuration either directly to the Arista EOS devices or to
Configlets within CloudVision. This is typically executed from the CLI - for example from your Ansible AVD Examples directory. 

### Using Ansible AVD with direct eAPI connectivity to the switches

If you want to push to switches in the entire FABRIC using eAPI and your playbook looks as follows:

```yml
---
- name: Run AVD
  hosts: FABRIC
  gather_facts: false
  tasks:
    - name: generate intended variables
      import_role:
        name: arista.avd.eos_designs
    - name: generate device intended config and documentation
      import_role:
        name: arista.avd.eos_cli_config_gen
    - name: deploy configuration to device
      import_role:
         name: arista.avd.eos_config_deploy_eapi
```

You would execute it using the following command:

```shell

> ansible-playbook playbook.yml

Reviewer 3: this assumes that the playbook file is called `playbook.yml` which may not be the case

Author: Added more clarity

```

This will:

- Create a text file containing the device's entire configuration.
- Create device-specific and fabric-wide documentation.
- Push the relevant configuration to each device using eAPI.

> ### Using Ansible AVD in accordance with CloudVision

Reviewer 2: ```suggestion
### Using Ansible AVD in conjunction with CloudVision
```

Author: Updated

        name: arista.avd.eos_designs
    - name: generate device intended config and documentation
      import_role:
        name: arista.avd.eos_cli_config_gen
    - name: deploy configuration to device
      import_role:
         name: arista.avd.eos_config_deploy_eapi

- name: Push to CVP
  hosts: CVP
  gather_facts: false
  connection: local

  tasks:
    - name: run CVP provisioning
      import_role:
        name: arista.avd.eos_config_deploy_cvp
      vars:
        container_root: 'DC1'
        configlets_prefix: 'AVD'
        execute_tasks: false
        apply_mode: strict
        state: present
        cv_collection: v3
```

You would execute it using the same command:

```shell
ansible-playbook playbook.yml
```

This will:

- Create a CloudVision Configlet for each device in the inventory, containing the device's entire configuration.
- Build out containers within CloudVision that represent the hierarchy of groups represented in the inventory.
- Create tasks to apply the new configuration to each device within the inventory.

At this point the changes can be reviewed, approved and executed in accordance with existing company requirements through CloudVision.

> 

Reviewer 3: Just a question more than expecting a change but with this new intro example there is a loss of info as compared to the previous ones regarding the tags build/provisision. is it expected? is this info somewhere else?

Author: You are right. This initial example is meant as the 2nd step for a complete Ansible AVD rookie, after reading the "Introduction to Ansible and AVD". More advanced options will come in later examples.

        svis:
          # SVI definitions.
          "11":
            # SVI Description
            name: VRF1_VLAN11
            enabled: true
            # IP anycast gateway to be used in the SVI in every leaf.
            ip_address_virtual: 10.10.11.1/24
          "12":
            name: VRF1_VLAN12
            enabled: true
            ip_address_virtual: 10.10.12.1/24
      VRF2:
        vrf_vni: 2
        svis:
          "21":
            name: VRF2_VLAN21
            enabled: true
            ip_address_virtual: 10.10.21.1/24
          "22":
            name: VRF2_VLAN22
            enabled: true
            ip_address_virtual: 10.10.22.1/24

    l2vlans:
      # These are pure L2 vlans. They do not have a SVI defined in the l3leafs and they will be bridged inside the VXLAN fabric
      "3401":
        name: L2_VLAN3401
      "3402":
        name: L2_VLAN3402

```

You would then describe the scope for where these network services should exist in the inventory.yml file:

```yml
NETWORK_SERVICES:
  children:
    DC1_L3_LEAVES:
    DC1_L2_LEAVES:
    DC2_L3_LEAVES:
    DC2_L2_LEAVES:
```

This specifies all switches in the fabric will be able to serve the new tenant.

> For sake of simplicity, let's say you defined just one pair of leaf switches that would serve this tenant.

Reviewer 3: Originally present but 

```suggestion
For the sake of simplicity, let's say you defined just one pair of leaf switches that would serve this tenant.
```

Author: Updated

> - With [Ansible Tower](https://docs.ansible.com/ansible/2.3/tower.html)

Reviewer 2: I would just get rid of the With... wording there.

 - Role documentation for eos_designs
 - Role documentation for eos_cli_config_get
 - RedHat Ansible Tower Documentation
 - etc

Also note that the eos_designs link currently points to the eos_cli_config_gen docs.

Author: Fixed
